clevertap-core/src/main/java/com/clevertap/android/sdk/network/NetworkManager.java

```diff
@@ -694,169 +566,160 @@ public boolean sendQueue(final Context context, final EventGroup eventGroup, fin
             return false;
         }
 
-        HttpsURLConnection conn = null;
-        try {
-            final String endpoint = getEndpoint(false, eventGroup);
+        EndpointId endpointId = EndpointId.fromEventGroup(eventGroup);
```

Comment:
please make sure exceptions are not thrown in these blocks as we moved try block

Reply:
Only getQueueHeader contains lines that can throw and its whole body is currently wrapped with catch(Throwable t). On a different note I don't know why we catch Throwable instead of Exception.

Reply:
That is legacy code, lets handle the catch block correctly as required.